server/src/metrics/metric-result.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+export abstract class MetricResult {
```

Comment:
I think this can actually be an interface: https://www.typescriptlang.org/docs/handbook/interfaces.html

Reply:
Yeah, an abstract class would let you provide some default implementation for a method, but until we find we actually need that, I'd probably just go with an interface.